servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -9764,14 +9809,17 @@ public void putGroupMembership(ResourceContext ctx, String domainName, String gr
 
         // add the member to the specified role
 
-        dbService.executePutGroupMembership(ctx, domainName, group, groupMember, auditRef);
+       dbService.executePutGroupMembership(ctx, domainName, group, groupMember, auditRef);
 
         // new group member with pending status. Notify approvers
 
         if (groupMember.getApproved() == Boolean.FALSE) {
             sendGroupMembershipApprovalNotification(domainName, domain.getDomain().getOrg(), groupName,
                     groupMember.getMemberName(), auditRef, principal.getFullName(), group);
         }
+
+        // TODO roy - is it ok to return GroupMember if Got GroupMemberShip
```

Comment:
so here we have a problem because in some of the methods mostly in the members methods, the function get a Membership Object but returns GroupMember or RoleMember object so what should we do?

Reply:
if the UI needs a different object type then it either needs to generate one itself based on the data returned or make another call to server to retrieve the updated object.